pkg/build/reaper/reaper_test.go

```diff
@@ -0,0 +1,194 @@
+	"reflect"
+	"testing"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	ktestclient "k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client/testclient"
+)
+
+func makeBuildConfig(version int, deleting bool) *buildapi.BuildConfig {
+	ret := &buildapi.BuildConfig{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:        "config",
+			Namespace:   "default",
+			Annotations: make(map[string]string),
+		},
+		Spec: buildapi.BuildConfigSpec{},
+		Status: buildapi.BuildConfigStatus{
+			LastVersion: version,
+		},
+	}
+	if deleting {
+		ret.Annotations[buildapi.BuildConfigDeletionAnnotation] = ""
+	}
+	return ret
+}
+
+func makeBuild(version int) buildapi.Build {
+	return buildapi.Build{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      fmt.Sprintf("build-%d", version),
+			Namespace: "default",
+			Labels:    map[string]string{buildapi.BuildConfigLabel: "config"},
+		},
+	}
+}
+
+func makeDeprecatedBuild(version int) buildapi.Build {
+	return buildapi.Build{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      fmt.Sprintf("build-%d", version),
+			Namespace: "default",
+			Labels:    map[string]string{buildapi.BuildConfigLabelDeprecated: "config"},
+		},
+	}
+}
+
+func makeBuildList(version int) *buildapi.BuildList {
+	if version%2 != 0 {
+		panic("version needs be even")
+	}
+	list := &buildapi.BuildList{}
+	for i := 1; i <= version; i += 2 {
+		list.Items = append(list.Items, makeBuild(i))
+		list.Items = append(list.Items, makeDeprecatedBuild(i+1))
+	}
+	return list
+}
+
+func newBuildListFake(objects ...runtime.Object) *testclient.Fake {
+	fake := testclient.NewSimpleFake(objects...)
+	fake.PrependReactor("list", "builds", func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
+		selector := action.(ktestclient.ListAction).GetListRestrictions().Labels
+		retList := &buildapi.BuildList{}
+		for _, obj := range objects {
+			list, ok := obj.(*buildapi.BuildList)
+			if !ok {
+				continue
+			}
+			for _, build := range list.Items {
+				if selector.Matches(labels.Set(build.Labels)) {
+					retList.Items = append(retList.Items, build)
+				}
+			}
+		}
+		return true, retList, nil
+	})
+	return fake
+}
+
+func actionsAreEqual(a, b ktestclient.Action) bool {
+	if reflect.DeepEqual(a, b) {
+		return true
+	}
+	// If it's an update action, we will take a better look at the object
+	if a.GetVerb() == "update" && b.GetVerb() == "update" &&
+		a.GetNamespace() == b.GetNamespace() &&
+		a.GetResource() == b.GetResource() &&
+		a.GetSubresource() == b.GetSubresource() {
+		ret := reflect.DeepEqual(a.(ktestclient.UpdateAction).GetObject(), b.(ktestclient.UpdateAction).GetObject())
+		return ret
+	}
+	return false
+}
+
+func TestStop(t *testing.T) {
+	notFound := func() runtime.Object {
```

Comment:
Borrowed from deploy. I used it at two places before, not sure when I removed the one. But I still like it this way. It does not need to clutter up test tables and can be easily reused if new tests are added. I don't see any harm in it in any case.